esp/bindings/SOAP/xpp/fxpp/FragmentedXmlPullParser.h

```diff
@@ -0,0 +1,2145 @@
+            {
+                if (hasPrefix(a.first.c_str(), "xmlns", false))
+                    appendXMLAttr(m_content, a.first.c_str(), a.second.c_str(), nullptr, true);
+            }
+            m_content.append("/>");
+        }
+        else
+        {
+            // The absence of tag and attribute data requires only the trigger.
+            appendXMLOpenTag(m_content, reserved::SYNTHESIZED_GROUP, nullptr, true, true);
+        }
+    }
+
+    virtual void append(const IXmlFragment* fragment)
+    {
+        m_members.emplace_back(fragment);
+    }
+};
+
+/**
+ * Instruct the fragment to omit the root property tree element name from the serialized text. An
+ * omitted element's attributes are also omitted, except for those declaring namespaces. Namespace
+ * declarations are always preserved for the benefit of the element's content.
+ *
+ * One might exclude the root element when the element is not part of the syntax included in the
+ * fragment. Consider the example of ESDL integration scripts. Scripts located in property trees
+ * using `Scripts`, `Transforms`, and `CustomRequestTransform` element names. Neither `Scripts`
+ * nor `Transforms` are part of the script syntax and must be explicitly detected and ignored to
+ * prevent script parsing errors. Exclusion from the fragment reduces the likelihood that these
+ * out-of-language elements are included in the fragment.
+ *
+ * One might also exclude the root element if its content is expected to duplicate the root data.
+ * Returning to the ESDL integration script example, the `Scripts` element has been assumed to
+ * contain an embedded fragment that is itself rooted by a `Scripts` element. Exclusion from the
+ * fragment can reduce unnecessary redundancy.
+ */
+#define FXPP_PTREE_OMIT_TAG        XmlFragmentRules(0x80)
+
+/**
+ * Implementation of IXmlFragmentGroup representing a single property tree node and its
+ * contents.
+ * 
+ * - If the node contains text, the text is assumed to be embedded markup represented by a
+ *   single group member fragment. The group content is the node name and attributes by
+ *   default, and the trigger with namespaces preserved if the tag is omitted.
+ * - If the node contains child elements and the tag is omitted, the group content contains
+ *   the serialization of the child elements. No trigger is included and no group members
+ *   are created.
+ * - If the node contains child elements and tag is not omitted, the group content contains
+ *   the serialization of the node, in its entirety, with no trigger and no group members.
+ * - If the node is empty and the tag is omitted, the group content a synthesized placeholder
+ *   with no group members. It is low overhead enabling any node to be represented without
+ *   pre-validating its content.
+ * - If the node is empty and the tag is not omitted, the group content is the group name
+ */
+class CPTreeXmlFragment : implements IXmlFragmentGroup, public CBaseXmlFragmentGroup
```

Comment:
As we discussed I think that in general while we may use a ptree at the top level, we would only ever want nested content to be loaded from child content that is already parsed as a ptree in cases needed for backward compatibility.  No new functionality should support that.  The reason being that we shouldn't have to worry about ordering issues in our content.